fineract-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/Loan.java

```diff
@@ -1389,6 +1387,12 @@ private void applyPeriodicAccruals(final Collection<LoanTransaction> accruals) {
         }
     }
 
+    private LocalDate getDateForRangeCalculation(LoanTransaction loanTransaction, boolean isChargeAccrualBasedOnSubmittedOnDate) {
```

Comment:
It looks better, however i was wondering what if the loan charge is not a specific due date charge? Would it be better to call `getEffectiveDueDate()` method which covers the proper due date for installment charge as well?

Reply:
Yes, getEffectiveDueDate() covers charge due dates properly, I will update the PR.